### PR TITLE
Adds authorization - sign in, sign up

### LINKED ISSUE #19: Implementation of sign up/in mechanism
* Consult with the server team in order to conclude proper mechanism
* Determine how to authenticate UE
* Provide software implementation

### PR DESCRIPTION
**Pull request by krecik1219**
* Activities for Sign In and Sign Up
 * AccessToken storing
 * AccessToken added to database requests headers

 Closes #19

### CODE REVIEW

**Review by krp97 — DISMISSED** [review]

File: app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package pl.ppiwd.exerciseanalyst.common.auth;
+
+public final class TokenStore {
+    private static final Object lock = new Object();
+    private static TokenStore instance = null;
```
Comment (pawelqk) [inline app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java:5]: why not simply = new TokenStore();?

File: app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            }
+        }
+        return tokenStore;
+    }
+
+    private TokenStore() {
+        wipeToken();
+    }
+
+    public void storeToken(final String accessToken) {
```
Comment (pawelqk) [inline app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java:27]: assuming that it should be thread-safe then this has to be synchronized

File: app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        this.accessToken = accessToken;
+    }
+
+    public String getAccessToken() {
+        return accessToken;
+    }
+
+    public void wipeToken() {
```
Comment (pawelqk) [inline app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java:35]: ditto

File: app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java (hunk 1 above)
Comment (krecik1219) [reply app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java:5]: Assuming thread safety

File: app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java (hunk 2 above)
Comment (krecik1219) [reply app/src/main/java/pl/ppiwd/exerciseanalyst/common/auth/TokenStore.java:27]: You are right!

**Review by ArfreeX — APPROVED** [review]

**Review by krp97 — APPROVED** [review]

**Pull request merged**